hmcts/pre-api
# S28 3099: Stop Live Event (MediaKind Implemenation)

### JIRA ticket(s)

- S28-3099


### Change description
- Add MK implementation for stopping live event
- Add extra code to automatically create the `EncodeToMp4` transform if it doesn't exist





> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- General setup steps:
  - Create a capture session using PUT /capture-sessions/{id} endpoint
  - Start a live event (using PUT /media-service/live-events/start/{captureSessionId}
  - Create a streaming endpoint + locator (either using endpoint if merged in at the time of testing or manually via the MK portal)
- Test: Create + run a CVP recording for a few mins and then use the /media-services/live-event/stop/{captureSessionId} endpoint to end the recording + create the asset. See no error response, in mk see relateed streaming endpoint, locator and live event are all stopped and deleted. Using GET /recordings?captureSessionId={captureSessionId} endpoint see a newly created (v1) recording. See capture session has been updated with finshed at/by values + status is RECORDING_AVAILABLE. In MK, see newly created asset stored in final storage account
- Test: Do NOT create a CVP recording. Call stop endpoint. See no error, see in mk portal that related streaming endpoint, locator and live event are stopped and deleted. Using /recordings?captureSessionId={captureSessionId} endpoint get no values in response. See capture session has been updaetd with finished at/by values + status is NO_RECORDING. 
- Test: Attempt to stop a live event for a captuer session that does not exist- see 404
- Test: Attempt to stop a live event for a capture session that has been deleted- see 404
- Test: Attempt to stopp a live event that has already been finished- see 200
- Test: Attempt to stop a live event that has not been started- see 200
- Test: Attempt to stop a live event that has status that is not RECORDING or STANDBY- see 400
- Test: Start live event and then manually stop it in mk, now call stop endpoint. See error and capture session has been updates with finished at/by and status = FAILURE
- Test: Delete the EncodeToMp4 transform on mk. Run test 2- see 200. Look in mediakind and see EncodeToMp4 has been created (with correct config). 
- Any other tests (?)
